requests/models.py

```diff
@@ -447,6 +451,15 @@ def prepare_body(self, data, files, json=None):
         if is_stream:
             body = data
 
+            if getattr(body, 'tell', None) is not None:
+                # Record the current file position before reading.
+                # This will allow us to rewind a file in the event
+                # of a redirect.
+                try:
+                    self._body_position = body.tell()
+                except (IOError, OSError):
+                    pass
```

Comment:
@Lukasa, thoughts on setting `_body_position` to a sentinel object here? We currently have a gap where `_body_position` will remain `None` when `tell` fails and won't trigger the exception in `rewind_body`. It seems hasty to raise the exception here since we aren't sure a redirect will ever occur.